src/InteractionModel.H

```diff
@@ -41,80 +42,178 @@ namespace ExaEpi
+
+    + Create bins of agents if not already created (see
+      #amrex::GetParticleBin, #amrex::DenseBins):
+      + The bin size is 1 cell
+      + #amrex::GetParticleBin maps a particle to its bin index
+      + amrex::DenseBins::build() creates the bin-sorted array of particle indices and
+        the offset array for each bin (where the offset of a bin is its starting location
+        in the bin-sorted array of particle indices).
+
+    + For each agent *i* in the bin-sorted array of agents:
+      + Find its bin and the range of indices in the bin-sorted array for agents in its bin
+      + If the agent is #Status::immune, do nothing.
+      + If the agent is #Status::infected with the number of days infected (RealIdxDisease::disease_counter)
+        less than the incubation length, do nothing.
+      + Else, for each agent *j* in the same bin:
+        + If the agent is #Status::immune, do nothing.
+        + If the agent is #Status::infected with the number of days infected (RealIdxDisease::disease_counter)
+          less than the incubation length, do nothing.
+        + Else if *i* is not infected and *j* is infected, compute probability of *i* getting infected
+          from *j* (see below).
+
+    Summary of how the probability of agent A getting infected from agent B is computed:
+    + Compute infection probability reduction factor from vaccine efficacy (#DiseaseParm::vac_eff)
+    + Within family - if their IntIdx::nborhood and IntIdx::family indices are same,
+      and the agents are at home:
+      + If B is a child, use the appropriate transmission probability (#DiseaseParm::xmit_child_SC or
+        #DiseaseParm::xmit_child) depending on whether B goes to school or not (#IntIdx::school)
+      + If B is an adult, use the appropriate transmission probability (#DiseaseParm::xmit_adult_SC or
+        #DiseaseParm::xmit_adult) depending on whether B works at a school or not (#IntIdx::school)
+*/
+template <typename AgentContainer, typename PTDType, typename CandidateFunc, typename BinaryInteractionFunc>
+void interactAgentsImpl(AgentContainer& agents, int bin_idx)
+{
+    BL_PROFILE("interactAgentsimpl");
+    int n_disease = agents.numDiseases();
+    IntVect bin_size = {AMREX_D_DECL(1, 1, 1)};
+    CandidateFunc isCandidate;
+    BinaryInteractionFunc binaryInteraction;
 
-                    AMREX_ALWAYS_ASSERT(np == bins_ptr->numItems());
+    for (int lev = 0; lev < agents.numLevels(); ++lev)
+    {
+#ifdef AMREX_USE_OMP
+#pragma omp parallel if (Gpu::notInLaunchRegion())
+#endif
+        for (MFIter mfi = agents.MakeMFIter(lev, TilingIfNotGPU()); mfi.isValid(); ++mfi) {
+            auto& ptile = agents.ParticlesAt(lev, mfi);
+            const auto& ptd = ptile.getParticleTileData();
+            const auto np = ptile.GetArrayOfStructs().numParticles();
+            auto& soa = ptile.GetStructOfArrays();
+            auto num_tiles = numTilesInBox(mfi.validbox(), true, bin_size);
+
+            int max_group = 0;
```

Comment:
I also find that sometimes `max_group` gets set to something huge, I think there is an overflow or something in `CensusData::assignTeachersAndWorkgroup`

Reply:
Can you remind me what dataset you used to trigger this?

Reply:
Hi @stevenhofmeyr - it was the `inputs_random_ca` example.